lib/takeaway.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'menu.rb'
+
+class Takeaway
+
+  attr_reader :total
+
+  def initialize(menu_klass = Menu)
+    @menu = menu_klass.new
+    @basket = Hash.new(0)
+    @total = 0
+  end
+
+  def menu
+    @menu.dishes.dup
+  end
+
+  def order(dish, quantity=1)
+    fail 'Sorry, that item is not on the menu' if menu.has_key?(dish) == false
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`. (https://github.com/bbatsov/ruby-style-guide#hash-key)